src/client/particles.cpp

```diff
@@ -627,7 +627,10 @@ const core::aabbox3df &ParticleBuffer::getBoundingBox() const
 			continue;
 
 		for (u16 j = 0; j < 4; j++)
-			box.addInternalPoint(m_mesh_buffer->getPosition(i * 4 + j));
+			if (i == 0 && j == 0)
```

Comment:
incorrect assumption: the first valid index will be zero

Reply:
Yep, good catch. I'll have to think about it a little more.

Reply:
Drat. Easy solution: Just add a `bool first` indicating whether this is the first vertex.

Side note: More generally I feel that many places where we have `aabb3df` should probably be `std::optional<aabb3df>`, where an empty optional represents the absence of a box (no points). There could be something like a helper method `aabb3df::extend(std::optional<aabb3df> maybe_box, v3f point)` to codify the "pattern" of replacing the empty box with the single point box and extending after that.

Reply:
surely initializing it to `{0,0,0, 0,0,0}` would be easier?

Reply:
Why not just call `box.reset(m_mesh_buffer->getPosition(0 * 4 + 0));` one time on top of the loops? That entirely eliminates the branch within this loop.

Reply:
> Why not just call `box.reset(m_mesh_buffer->getPosition(0 * 4 + 0));` one time on top of the loops? That entirely eliminates the branch within this loop.

Due to the `continue` above the first vertex ID need not be valid.

If we did want to prematurely optimize the branch away (which is probably insignificant compared to `addInternalPoint`), we'd want to hoist the bool out of the loop: As soon as the first valid vertex is encountered, jump to a second version of the loop which does not contain the check. To avoid code duplication a `template<bool first>` function could be used.

But I don't think it's worth it. Just add a `bool` and be done with it. (And I say this as a hater of bools; I try to avoid them using proper control flow whenever I can.)